quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/Picocli.java

```diff
@@ -543,4 +551,40 @@ private static boolean isRuntimeOption(String arg) {
 
         return arg.startsWith(ImportRealmMixin.IMPORT_REALM);
     }
+
+    private static void checkChangesInBuildOptionsDuringAutoBuild() {
+        if (Configuration.isOptimized()) {
+            List<PropertyMapper> buildOptions = stream(Configuration.getPropertyNames(true).spliterator(), false)
+                    .sorted()
+                    .map(PropertyMappers::getMapper)
+                    .filter(Objects::nonNull).collect(Collectors.toList());
+
+            if (buildOptions.isEmpty()) {
+                return;
+            }
+
+            StringBuilder output = new StringBuilder("You run a 'build' before starting the server, the following build options will be overridden to new values. " +
+                    "To avoid that, run the 'build' command again and start the server using the '").append(OPTIMIZED_BUILD_OPTION_LONG).append("' flag:");
```

Comment:
What about this:

```
The previous optimized build will be overridden. Run the 'build' command again:

    kc.sh build <options previously set>

And start the server using the '--optimized' flag:

    kc.sh start --optimized 
```

Reply:
A bit more lengthy but perhaps more self-explanatory and with hints about how to properly use optimized server images?

Reply:
Maybe
```
The previous optimized build will be overridden with the following config values.

   - <current> > <new>

To avoid that, run the 'build' command again:

    kc.sh build <options previously set>

Then start the optimized server instance using the '--optimized' flag:

    kc.sh start --optimized 
```

Reply:
Works for me. I would just replace `config values` with `build options`.

Reply:
Final:

```
The previous optimized build will be overridden with the following build options:
        - cache=local > cache=ispn
        - db=postgres > db=dev-file
        - features=preview > features=<unset>
To avoid that, run the 'build' command again and then start the optimized server instance using the '--optimized' flag.
```